overlord/snapstate/snapstate.go

```diff
@@ -209,6 +209,54 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 	return doInstall(s, snapst.Active, ss)
 }
 
+// Enable sets a snap to the active state
+func Enable(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	if snapst.Active {
+		return nil, fmt.Errorf("snap %q already enabled", name)
+	}
+
+	ss := &SnapSetup{
+		Name:     name,
+		Revision: snapst.Current,
+	}
+
+	revisionStr := fmt.Sprintf(" (%d)", snapst.Current)
+	prepareSnap := s.NewTask("prepare-snap", fmt.Sprintf(i18n.G("Prepare snap %q%s"), ss.Name, revisionStr))
+	prepareSnap.Set("snap-setup", &ss)
+
+	linkSnap := s.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q available to the system%s"), ss.Name, revisionStr))
+	linkSnap.Set("snap-setup", &ss)
+	linkSnap.WaitFor(prepareSnap)
+
+	return state.NewTaskSet(prepareSnap, linkSnap), nil
+}
+
+// Disable sets a snap to the inactive state
+func Disable(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	if !snapst.Active {
+		return nil, fmt.Errorf("snap %q already disabled", name)
+	}
+
+	ss := &SnapSetup{
+		Name: name,
+	}
+
+	unlinkSnap := s.NewTask("unlink-current-snap", fmt.Sprintf(i18n.G("Make current revision for snap %q unavailable (%d)"), ss.Name, snapst.Current))
```

Comment:
cannot this use unlink-snap instead? if we really need this one, now it would be used standalone and needs to be added to the ones checkChangeConflict cares about